readthedocs/doc_builder/python_environments.py

```diff
@@ -197,8 +200,39 @@ def is_obsolete(self):
             env_python_version != self.config.python_full_version,
             env_build_image != build_image,
             env_build_hash != image_hash,
+            env_vars_hash != self._get_env_vars_hash(),
         ])
 
+    def _get_env_vars(self):
+        """
+        Returns environment variables with their values of the associated project.
+
+        If there are no environment variables, it returns None.
+        """
+        env_vars = self.version.project.environmentvariable_set.values_list('name', 'value')
+        if env_vars.exists():
+            return tuple(env_vars)
+
+    def _get_env_vars_hash(self):
+        """
+        Returns the sha256 hash of all the environment variables and their values.
+
+        If there are no environment variables, it returns the sha256 hash of 'None'.
+        """
+        env_vars = self._get_env_vars()
+        m = hashlib.sha256()
+        if env_vars:
+            for env_var in env_vars:
+                hash_str = '_{var}_{value}_'.format(
+                    var=env_var[0],
+                    value=env_var[1]
+                )
+                m.update(hash_str.encode('utf-8'))
+        else:
+            m.update('None'.encode('utf-8'))
```

Comment:
No need to add this magic constant, I think we can just rely on the default value of the hash `b''` (the default)